src/btf.cpp

```diff
@@ -722,8 +725,11 @@ std::pair<int, int> BTF::get_btf_id_fd(const std::string &func,
       continue;
 
     auto id = find_id_in_btf(btf_obj.btf, func, BTF_KIND_FUNC);
-    if (id >= 0)
+    if (id >= 0) {
+      if (btf_obj.id == 0)
+        return { id, 0 };
       return { id, bpf_btf_get_fd_by_id(btf_obj.id) };
```

Comment:
Perhaps more compact variant:
```suggestion
      return { id, btf_obj.id ? bpf_btf_get_fd_by_id(btf_obj.id) : 0 };
```